src/hlist.rs

```diff
@@ -12,6 +18,7 @@ pub struct Cons<T, U = Nil> {
 }
 
 impl Nil {
+    /// Add an element to the list
```

Comment:
I would use a more explicit form:
```suggestion
    /// Prepend a new _head_ to the heterogeneous list
```

Reply:
Folded in locally as well.